pkgs/games/katago/default.nix

```diff
@@ -0,0 +1,100 @@
+{ stdenv
+, gcc8Stdenv
+, lib
+, libzip
+, boost
+, cmake
+, makeWrapper
+, fetchFromGitHub
+, linuxPackages ? null
+, cudnn ? null
+, cudatoolkit ? null
+, libGL_driver ? null
+, opencl-headers ? null
+, ocl-icd ? null
+, gperftools ? null
+, cudaSupport ? false
+, useTcmalloc ? false}:
+
+assert !cudaSupport || (
+  linuxPackages != null && 
+  libGL_driver != null && 
+  cudatoolkit != null &&
+  cudnn != null);
+
+assert cudaSupport || (
+  opencl-headers != null &&
+  ocl-icd != null);
+
+assert !useTcmalloc ||
+  gperftools != null;
+
+let
+  env = if cudaSupport 
+    then gcc8Stdenv
+    else stdenv;
+
+in env.mkDerivation rec {
+  pname = "katago";
+  version = "1.3.3";
+
+  src = fetchFromGitHub {
+    owner = "lightvector";
+    repo = "katago";
+    rev = "v${version}";
+    sha256 = "031pq0lhv2q46iip1d6nrhx5r669l3jc7gzdpw74j9xbhjvza9kz";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    makeWrapper
+  ];
+
+  buildInputs = [
+    libzip
+    boost
+  ] ++ lib.optionals cudaSupport [
+    linuxPackages.nvidia_x11
+    cudnn
+    libGL_driver
+  ] ++ lib.optionals (!cudaSupport) [
+    opencl-headers
+    ocl-icd
+  ] ++ lib.optionals useTcmalloc [
+    gperftools
+  ];
+
+  cmakeFlags = [
+    "-DNO_GIT_REVISION=ON"
+  ] ++ lib.optionals cudaSupport [
+    "-DUSE_BACKEND=CUDA"
+  ] ++ lib.optionals (!cudaSupport) [
+    "-DUSE_BACKEND=OPENCL"
+  ] ++ lib.optionals useTcmalloc [
+    "-DUSE_TCMALLOC=ON"
+  ];
+
+  preConfigure = ''
+    cd cpp/
+  '' + lib.optionalString cudaSupport ''
+    export CUDA_PATH="${cudatoolkit}"
+    export EXTRA_LDFLAGS="-L${linuxPackages.nvidia_x11}/lib"
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin; cp katago $out/bin;
+  '' + lib.optionalString cudaSupport ''
+    wrapProgram $out/bin/katago \
+      --prefix LD_LIBRARY_PATH : "${linuxPackages.nvidia_x11}/lib"
```

Comment:
I'm not sure about this and `EXTRA_LDFLAGS` above, because it directly references a specific build, but maybe someone is using nvidiaBeta/Legacy or a different kernel version, and these libraries won't work. In that case, this should rather be `/run/opengl-driver/lib`.

Reply:
Good catch.  I pulled this from @grahamc 's Nix CUDA example github.  I'm pretty sure if you're using legacy drivers for nvidia you probably won't be able to run CUDA anyway (and/or running CUDA will have no advantages over plain OpenCL).

But for the beta version or a different kernel version, will the drivers in `/run/opengl-driver/lib` be sufficient?  I've tested it out on my computer and it works with the default versions on my computer.

Reply:
I don't know, I'm also on default. Also the package won't get rebuilt if the drivers are upgraded, so I'm not sure. Maybe this is the recommended way.

Reply:
I had a talk with Clever on IRC about it, and he agreed that it should be `/run/opengl-driver/lib`; moreover, the rest of the packages in nixpkgs seem to also follow this format.